nipap/nipap/authlib.py

```diff
@@ -327,17 +335,45 @@ def authenticate(self):
 
         # auth succeeded
         self.authenticated_as = self.username
-        self._authenticated = True
         self.trusted = False
         self.readonly = False
 
         try:
-            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, 'uid=' + self.username, ['cn'])
+            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, self._ldap_search.format(ldap.dn.escape_dn_chars(self.username)), ['cn','memberOf'])
```

Comment:
This search should not be performed if both ro_group and rw_group are empty. That would make it possible to use empty ro_group/rw_group to provide everyone with read/write access if you don't have the memberOf attribute in your LDAP instance.

Currently, the search_s call would raise an NO_SUCH_ATTRIBUTE exception, meaning all authenticated users would get read only access instead.